### PR TITLE
New Exercise: Pascal's Triangle

### PR DESCRIPTION
**Pull request by jvarness**
Pascal's Triangle!

### CODE REVIEW

**Comment by jvarness** [comment]
@Stargator @devkabiir lemme know what you think

File: exercises/pascals-triangle/lib/example.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class PascalsTriangle {
+  List<List<num>> rows(int rows) {
```
Comment (devkabiir) [inline exercises/pascals-triangle/lib/example.dart]: Does it have to be generic `num`? Can we not use `int`

Comment (jvarness) [reply exercises/pascals-triangle/lib/example.dart]: No, it doesn't have to be `num`, I just think that's how it got generated. I'll make a note to change it.

**Comment by jvarness** [comment]
Does the config look ok to you guys? Correct difficulty? Correct topics? Needs to be `unlocked_by` something or not?

**Comment by Stargator** [comment]
You should run presubmit

**Comment by jvarness** [comment]
@Stargator presubmit didn't seem to make any suggestions regarding the config for either diamond or pascal

**Comment by Stargator** [comment]
@jvarness Just checking in, I hope you're doing well. If no changes are done by February. I may start looking at submitting the changes myself.

**Comment by jvarness** [comment]
Hey @Stargator sorry about that. I'm doing well :)

No need, I'll make the changes when I find some spare time.

File: exercises/pascals-triangle/test/pascals_triangle_test.dart [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+              [1, 6, 15, 20, 15, 6, 1],
+              [1, 7, 21, 35, 35, 21, 7, 1],
+              [1, 8, 28, 56, 70, 56, 28, 8, 1],
+              [1, 9, 36, 84, 126, 126, 84, 36, 9, 1]
+            ]));
+      }, skip: true);
+
+      test("negative rows", () {
+        final List<List<int>> result = pascalsTriangle.rows(-1);
+        expect(result, equals(<List<int>>[]));
```
[the remark this answers is not in the document]
Comment (jvarness) [reply exercises/pascals-triangle/test/pascals_triangle_test.dart]: I removed this test from the fixture.

File: exercises/pascals-triangle/lib/example.dart (hunk 1 above)
Comment (jvarness) [reply exercises/pascals-triangle/lib/example.dart]: Changed everything to `int`

File: exercises/pascals-triangle/test/pascals_triangle_test.dart [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import 'package:test/test.dart';
+import 'package:pascals_triangle/pascals_triangle.dart';
+
+void main() {
+  final pascalsTriangle = new PascalsTriangle();
+
+  group('PascalsTriangle', () {
+    group('Given a count, return a collection of that many rows of pascal\'s triangle', () {
```
[the remark this answers is not in the document]
Comment (jvarness) [reply exercises/pascals-triangle/test/pascals_triangle_test.dart]: Corrected.

File: exercises/pascals-triangle/test/pascals_triangle_test.dart [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      test("zero rows", () {
+        final List<List<int>> result = pascalsTriangle.rows(0);
+        expect(result, equals(<Object>[]));
```
[the remark this answers is not in the document]
Comment (jvarness) [reply exercises/pascals-triangle/test/pascals_triangle_test.dart]: Corrected.

**Comment by jvarness** [comment]
@Stargator rebased this with master and made all the corrections you and @devkabiir have requested.

**Comment by Stargator** [comment]
@jvarness Just rebase your changes ASAP and you can merge it in! 💯 

Sorry for the long wait.

**Pull request merged**